test-tools/aot-analyzer/src/main.cc

```diff
@@ -0,0 +1,558 @@
+    parser.AddOption('c', "compare",
+                     "Show the differences between AoT and WASM",
+                     []() { s_objdump_options.compare = true; });
+    parser.AddArgument(
+        "filename", OptionParser::ArgumentCount::OneOrMore,
+        [](const char *argument) { s_infiles.push_back(argument); });
+    parser.Parse(argc, argv);
+}
+
+void
+InitStdio()
+{
+#if COMPILER_IS_MSVC
+    int result = _setmode(_fileno(stdout), _O_BINARY);
+    if (result == -1) {
+        perror("Cannot set mode binary to stdout");
+    }
+    result = _setmode(_fileno(stderr), _O_BINARY);
+    if (result == -1) {
+        perror("Cannot set mode binary to stderr");
+    }
+#endif
+}
+
+void
+DumpInfo(AoTFile *aot)
+{
+    const AOTTargetInfo target_info = aot->GetTargetInfo();
+    printf("AOT File Information: \n\n");
+    printf("Binary type: %s \n",
+           aot->GetBinTypeName(target_info.bin_type).c_str());
+    printf("ABI type: %d \n", target_info.abi_type);
+    printf("Exectuion type: %s \n",
+           aot->GetExectuionTypeName(target_info.e_type).c_str());
+    printf("Exectuion machine: %s \n",
+           aot->GetExectuionMachineName(target_info.e_machine).c_str());
+    printf("Exectuion version: %d \n", target_info.e_version);
+    printf("Exectuion flags: %d \n", target_info.e_flags);
+    printf("Feature flags: %ld \n", target_info.feature_flags);
+    printf("Reserved: %ld \n", target_info.reserved);
+    printf("Arch: %s \n", target_info.arch);
+}
+
+void
+DumpTextSize(AoTFile *aot)
+{
+    const AOTTargetInfo target_info = aot->GetTargetInfo();
+    printf("%s:       file format <%s> \n\n", aot->GetFileName(),
+           aot->GetBinTypeName(target_info.bin_type).c_str());
+    printf("Text size: \n");
+
+    const uint32_t literal_size =
+        ((AOTModule *)(aot->GetModule()))->literal_size;
+    const uint32_t code_size = ((AOTModule *)(aot->GetModule()))->code_size;
+    printf("   literal size= %d Bytes \n", literal_size);
+    printf("      code size= %d Bytes \n", code_size);
+}
+
+void
+DumpDetails(AoTFile *aot)
+{
+    const AOTTargetInfo target_info = aot->GetTargetInfo();
+    printf("%s:          file format <%s> \n\n", aot->GetFileName(),
+           aot->GetBinTypeName(target_info.bin_type).c_str());
+    printf("Details: \n\n");
+
+    // Types
+    const uint32_t type_count = ((AOTModule *)(aot->GetModule()))->type_count;
+    AOTType **types = ((AOTModule *)(aot->GetModule()))->types;
+    printf("Types[%d] \n", type_count);
+
+    const char *wasm_type[] = { "function", "struct", "array" };
+    for (uint32_t index = 0; index < type_count; index++) {
+        AOTType *type = types[index];
+        const uint16_t type_flag = type->type_flag;
+        if (type_flag == WASM_TYPE_FUNC) {
+            printf("  -[%d] %s    param_count:%d    result_count:%d \n", index,
+                   wasm_type[type_flag], ((AOTFuncType *)type)->param_count,
+                   ((AOTFuncType *)type)->result_count);
```

Comment:
Will we dump detailed reference type for each param/result type? Same for struct's field type, array's elem type, table elem type, table segment's element type, global type and so on.

Reply:
Outputting these details does seem a bit redundant

Reply:
done